src/tools/wasm-ctor-eval.cpp

```diff
@@ -66,8 +72,9 @@ class EvallingGlobalManager {
     if (dangerousGlobals.count(name) > 0) {
       std::string extra;
       if (name == "___dso_handle") {
-        extra = "\nrecommendation: build with -s NO_EXIT_RUNTIME=1 so that "
-                "calls to atexit that use ___dso_handle are not emitted";
+        extra = RECOMMENDATION
+          "build with -s NO_EXIT_RUNTIME=1 so that "
+          "calls to atexit that use ___dso_handle are not emitted";
```

Comment:
Unrelated to this change, but I'm not sure this is true anymore is it?  

Today NO_EXIT_RUNTIME means that atexit is a dummy function but IIRC that calls to atexit are not effected and wasm-opt/LTO is requires to clean them up.

Reply:
Oh, right, your recent change to atext is relevant here. I think this code is not needed actually. I can remove it in a followup.